### PR TITLE
ENT-140: Allow catalog to be disabled for EnterpriseCustomer

### PR DESCRIPTION
**Pull request by haikuginger**
**Description:**

This pull request adds a migration and code to ensure that when saving an EnterpriseCustomerAdminForm, a user may select a catalog, save the object, and then later remove that catalog from being linked to the EnterpriseCustomer.

**JIRA:** [ENT-140](https://openedx.atlassian.net/browse/ENT-140)

**Dependencies:** None

**Merge deadline:** 30 January 2017

**Testing instructions:**

1. With the existing code, and with a working Course Discovery devstack, create an enterprise customer, and select a catalog from the Course Catalog dropdown menu.
2. Save the enterprise customer
3. Attempt to switch the enterprise customer back to using no catalog
4. Save the enterprise customer
5. Observe that the enterprise customer keeps its existing catalog
6. Switch to this code.
7. Attempt to switch the enterprise customer back to using no catalog
8. Save the enterprise customer
9. Observe that the change has been saved.

**Reviewers:**
- [x] @bdero  
- [ ] edX TBD 

**Merge checklist:**
- [x] All reviewers approved
- [x] CI build is green
- [x] Version bumped
- [x] Changelog record added
- [x] Commits are (reasonably) squashed
- [x] Translations are updated
- [x] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @haikuginger! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

<a   href="https://openedx-webhooks.herokuapp.com/github/process_pr?number=31&repo=edx%2Fedx-enterprise"   rel="nofollow" >Create an OSPR issue for this pull request</a>.

**Comment by bdero** [comment]
@haikuginger Is there a course discovery API out in the open I can use somewhere, or do I need to set something up? I found references in other PRs to https://discovery-enterprise.sandbox.edx.org/, but keeping in mind that I have no context for this, could you give me some directions?

**Comment by bradenmacdonald** [comment]
@bdero Instructions for setting it up locally are at https://github.com/edx/course-discovery/blob/master/docs/getting_started.rst . That said, discovery-enterprise.sandbox.edx.org was working and very helpful but seems to be offline at the moment... @haikuginger do you know why?

File: tests/admin/test_forms.py [hunk 1]
```diff
@@ -281,7 +281,7 @@ def test_interface_displays_selected_option(self, mock_method):
         form = EnterpriseCustomerAdminForm()
         assert isinstance(form.fields['catalog'], forms.ChoiceField)
         assert form.fields['catalog'].choices == [
-            (None, 'None'),
+            ('', '---------'),
```
Comment (bdero) [inline tests/admin/test_forms.py]: Nit: Can we make this use `django.db.models.fields.BLANK_CHOICE_DASH` instead?

File: tests/admin/test_forms.py [hunk 2]
```diff
@@ -303,7 +303,7 @@ def test_with_mocked_get_edx_data(self, mocked_get_edx_data, *args):  # pylint:
...
```
Comment (bdero) [inline tests/admin/test_forms.py]: Nit: Same comment about BLANK_CHOICE_DASH here.

**Comment by bdero** [comment]
@haikuginger I got it working and was able to test this - I only have one nit comment. :+1:

- [x] I tested this: I setup the course-discovery stack and confirmed that Enterprise Customer catalogs can't be reset to None once set (even though the choice is there). Then I installed this version (`pip install -e git+https://github.com/open-craft/edx-enterprise.git@haikuginger/nullable-catalog#egg=edx-enterprise==catalog-test && ./manage.py lms migrate --settings=devstack`) and confirmed that it can be set blank now.
- [x] I read through the code
- [ ] ~~I checked for accessibility issues~~ Not necessary, this change involves no UX modifications (other than showing dashes instead of "None" in the Catalog dropdown).
- [ ] ~~Includes documentation~~ Not necessary, this is a small bug fix.

**Comment by haikuginger** [comment]
@bdero, done.

@mattdrayer, @asadiqbal08, @saleem-latif, ready for review.

**Comment by mattdrayer** [comment]
:+1: for me!

**Comment by saleem-latif** [comment]
LGTM 👍

**Comment by asadiqbal08** [comment]
👍

**Comment by haikuginger** [comment]
Rebased from 8bdfd0d.

**Pull request merged**